tests/conftest.py

```diff
@@ -89,3 +86,116 @@ def _snapshot(**kwargs):
             yield snapshot
 
     return _snapshot
+
+
+if PY2:
+    import marshal
+    from py_compile import MAGIC
+    from py_compile import wr_long
+
+    from _pytest._code.code import ExceptionInfo
+
+    def dump_code_to_file(code, file):
+        file.write(MAGIC)
+        wr_long(file, long(time.time()))  # noqa
+        marshal.dump(code, file)
+        file.flush()
+
+
+else:
+    import importlib
+
+    code_to_pyc = getattr(
+        importlib._bootstrap_external, "_code_to_bytecode" if sys.version_info < (3, 7) else "_code_to_timestamp_pyc"
+    )
+
+    def dump_code_to_file(code, file):
+        file.write(code_to_pyc(code, time.time(), len(code.co_code)))
+        file.flush()
+
+
+def unwind_params(params):
+    if params is None:
+        yield None
+        return
+
+    for _ in product(*(((k, v) for v in params[k]) for k in params)):
+        yield dict(_)
+
+
+def run_function_from_file(item, params=None):
```

Comment:
can we add a docstring here?

Reply:
or wherever the best place to document the marker, how to use it, etc

Reply:
👍  I've improved the docstring in the marker registration in the root `conftest.py`